File: selectpointlabel.cpp
#include "selectpointlabel.h"

SelectPointLabel::SelectPointLabel(QWidget *parent):QLabel(parent)
{

}

void SelectPointLabel::mousePressEvent(QMouseEvent *event){
    emit mousePressed(event->pos().x(),event->pos().y());
}

void SelectPointLabel::resizeEvent(QResizeEvent * e)
{
    if(this->pixmap()!=0)
    this->setPixmap(this->pixmap()->scaledToHeight(this->height()));
    QLabel::resizeEvent(e);

}

File: selectpointlabel.py
from PyQt5.QtWidgets import QLabel
from PyQt5.QtCore import pyqtSignal

class SelectPointLabel(QLabel):
    mousePressed=pyqtSignal(int, int)
    
    def __init__(self, parent=None):
        super().__init__()
    
    def mousePressEvent(self, e):
        self.mousePressed.emit(e.pos().x(), e.pos().y())
